Test/Core/Utility/CoroutinesTest.cs

```diff
@@ -0,0 +1,195 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Duality.Resources;
+using NUnit.Framework;
+
+namespace Duality.Tests.Utility
+{
+	[TestFixture]
+	public class CoroutinesTest
```

Comment:
I think we should try to start to write the unit tests with the A(rrange) A(ct) A(ssert) pattern. You might have to split some unit tests to do this but this is a good thing as it makes the test more focused.

https://www.c-sharpcorner.com/UploadFile/dacca2/fundamental-of-unit-testing-understand-aaa-in-unit-testing/

Reply:
Agreed, but in this case I believe you should test the evolution of the Coroutine, and not only its final state, that's why I did them like that.